### ISSUE TITLE
clippy::all is not all

### ISSUE THREAD

**Issue by BenjaminBrienen** [rust-lang/rust-clippy#15108]
### Summary

In my build script, I generate code. I put attributes on that code to silence lints.

### Reproducer

I tried this code:

```rust
#![allow(clippy::all, reason = "generated code")]
#![allow(rustc::all, reason = "generated code")]

enum X {
    A,
    B,
    C,
}

impl X {
    fn foo(self) {
        match self {
            X::A => todo!(),
            X::B => todo!(),
            X::C => todo!(),
        }
    }
}
```

I expected to see this happen: no lints triggered

Instead, this happened:

A bunch, such as:
```
warning: this ident consists of a single char
 --> crates/profile/src/lib.rs:4:6
  |
4 | enum X {
  |      ^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#min_ident_chars
  = note: `-W clippy::min-ident-chars` implied by `-W clippy::restriction`
  = help: to override `-W clippy::restriction` add `#[allow(clippy::min_ident_chars)]`
```

### Version

```text

```

**Comment by profetia** [comment]
The `all` is very misleading and there is already discussions about renaming it.

**Comment by blyxyas** [comment]
See https://rust-lang.zulipchat.com/#narrow/channel/257328-clippy/topic/Renaming.20clippy.3A.3Aall/with/510794361

**Comment by samueltardieu** [comment]
I'm closing the issue as this has been in discussion, and in any case `clippy::all` will neither be deprecated nor get a change of scope. We will only recommend the use of another name (`clippy::default` or something else) everywhere in the documentation.
